src/libraries/System.Private.CoreLib/src/System/Number.NumberBuffer.cs

```diff
@@ -70,6 +70,28 @@ public void CheckConsistency()
 
                 Debug.Assert(numDigits == DigitsCount, "Null terminator found in unexpected location in Number");
                 Debug.Assert(numDigits < Digits.Length, "Null terminator not found in Number");
+
+                uint totalDigits = (uint)(DigitsCount);
+                uint positiveExponent = (uint)(Math.Max(0, Scale));
+                uint integerDigitsPresent = Math.Min(positiveExponent, totalDigits);
+                uint fractionalDigitsPresent = totalDigits - integerDigitsPresent;
+
+                // Verify that the fractional part does not have any trailing zeros
```

Comment:
Just wanting to confirm, we only need to do this for the fractional part and not the integral part?

That is `5000` is fine and should not be `5` with the right scale?

Reply:
Based on the loop below, it looks like we are just trimming all trailing zeros, which is why I ask.

Reply:
> ust wanting to confirm, we only need to do this for the fractional part and not the integral part?

Correct. We only work on the last `fractionalDigitsPresent` number of fractional digits in the buffer.

> it looks like we are just trimming all trailing zeros

Hmm, I don't think so. We're trimming away all trailing zeros only from the fractional part.